lib/workers/repository/updates/generate.ts

```diff
@@ -13,6 +13,33 @@ import { uniq } from '../../../util/uniq';
 import type { BranchConfig, BranchUpgradeConfig } from '../../types';
 import { CommitMessage } from '../model/commit-message';
 
+function mergeProxy(
```

Comment:
How come no existing merge logic such as mergeChildConfig() wasn't suitable?

Reply:
`templateInput` is `upgrade` + `{ upgrades: ... }`. One of the "inputs" (`upgrade`) changes when `templateInput` already exists. So it's not enough to just merge once - either need a proxy, or have to merge multiple times, after any value in `upgrade` gets set/updated.

Reply:
At this point I feel that this extra complexity is not justified by the benefits. Let's go back to the Discussion instead

Reply:
https://github.com/renovatebot/renovate/discussions/29546

But I guess the only alternative is "you can't do that"?

Reply:
That might be the case. I know it can be hard to hear, but at times we have to say "Sorry, the complexity you are suggesting is not justified by the benefit to one or a small number of users". Let's discuss there.